### PR TITLE
Issue/3553 test http

### PR DESCRIPTION
**Pull request by e-carlin**
Cleans things up and solves a cancel problem in test_http.

### CODE REVIEW

File: sirepo/auth/__init__.py [hunk 1]
```diff
@@ -606,7 +606,10 @@ def _auth_state():
         v.roles = auth_db.UserRole.get_roles(u)
         _plan(v)
         _method_auth_state(v, u)
-    if pkconfig.channel_in('dev'):
+    # TODO(e-carlin): discuss with rn. Ok to leak uid on alpha? Could parse cookie...
+    # Not strictly necessary but I think makes debuggin easier by having the uid
+    # in the test_http logs
+    if pkconfig.channel_in_internal_test():
```
Comment (e-carlin) [inline sirepo/auth/__init__.py:612]: @robnagler thoughts?

Comment (robnagler) [reply sirepo/auth/__init__.py:612]: Probably fine.

File: sirepo/pkcli/test_http.py [hunk 2]
```diff
@@ -152,6 +152,8 @@ async def _sims_tasks():
                 s.append(
                     _Sim(a, e.name, r.report, r.binary_data_file).create_task(),
                 )
+                # Sleep a bit to not completely flood server
+                await tornado.gen.sleep(1)
```
Comment (robnagler) [inline sirepo/pkcli/test_http.py:156]: shouldn't the server just queue?

File: sirepo/pkcli/test_http.py [hunk 3]
```diff
@@ -190,15 +192,19 @@ async def get_sim(self, sim_name):
...
             self.__class__.__name__,
             self.sim_type,
             self.client.email,
+            self.client.uid,
         )
 
 
 async def _cancel_all_tasks(tasks):
     for t in tasks:
+        # Sleep a bit to not completely flood server
```
Comment (robnagler) [inline sirepo/pkcli/test_http.py:205]: document this as, say, `await pause_for_server()`

File: sirepo/pkcli/test_http.py [hunk 4]
```diff
@@ -211,6 +217,7 @@ class _Client(PKDict):
     def __init__(self, email, **kwargs):
         super().__init__(
             email=email,
+            uid='<unknown>',
```
Comment (robnagler) [inline sirepo/pkcli/test_http.py:220]: why not None?

File: sirepo/pkcli/test_http.py [hunk 5]
```diff
@@ -378,18 +390,20 @@ def create_task(self):
         async def _run():
             _sims.append(self)
             # Must be set here once we are in the _run() task
-            self._task_id = str(id(asyncio.current_task()))[-4:]
+            self._task_id = _task_id(asyncio.current_task())
```
Comment (robnagler) [inline sirepo/pkcli/test_http.py:393]: `_current_task_id` would be a better encapsulation.

**Review by robnagler — APPROVED** [review]

**Comment by e-carlin** [comment]
Fix #3553

File: sirepo/pkcli/test_http.py (hunk 5 above)
Comment (e-carlin) [reply sirepo/pkcli/test_http.py:393]: But [it isn't always the current task](https://github.com/radiasoft/sirepo/pull/3557/files#diff-0b269c6d8875a46dc3afb0eb98ebc99343ccda7ad3346e971bd936e842d305feR207)

File: sirepo/auth/__init__.py (hunk 1 above)
Comment (e-carlin) [reply sirepo/auth/__init__.py:612]: https://github.com/radiasoft/sirepo/pull/3560/commits/bb28d6519751b4accde6307e65b4b58e80b6fbde

File: sirepo/pkcli/test_http.py (hunk 2 above)
Comment (e-carlin) [reply sirepo/pkcli/test_http.py:156]: It does but nginx in dev was just closing the connection when too many requests came in too quickly

https://github.com/radiasoft/sirepo/pull/3560/commits/f1f9b1aa4be4c77396e3467d5038c7b700e3997a

File: sirepo/pkcli/test_http.py (hunk 3 above)
Comment (e-carlin) [reply sirepo/pkcli/test_http.py:205]: https://github.com/radiasoft/sirepo/pull/3560/commits/f1f9b1aa4be4c77396e3467d5038c7b700e3997a

File: sirepo/pkcli/test_http.py (hunk 4 above)
Comment (e-carlin) [reply sirepo/pkcli/test_http.py:220]: 41fdca83

**Pull request merged**